logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/GetOrCreateOneToOneConversationUseCase.kt

```diff
@@ -1,33 +1,37 @@
 package com.wire.kalium.logic.feature.conversation
 
 import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.StorageFailure
 import com.wire.kalium.logic.data.conversation.Conversation
 import com.wire.kalium.logic.data.conversation.ConversationRepository
 import com.wire.kalium.logic.data.conversation.Member
 import com.wire.kalium.logic.data.user.UserId
-import com.wire.kalium.logic.functional.Either
-import com.wire.kalium.logic.functional.flatMap
 import com.wire.kalium.logic.functional.fold
 
 class GetOrCreateOneToOneConversationUseCase(
     private val conversationRepository: ConversationRepository,
 ) {
 
-    suspend operator fun invoke(otherUserId: UserId): CreateConversationResult =
-        conversationRepository.getOneToOneConversationDetailsByUserId(otherUserId).flatMap { conversation ->
-            if (conversation != null) {
-                Either.Right(conversation.conversation)
-            } else {
-                conversationRepository.createGroupConversation(members = listOf(Member(otherUserId)))
-            }
-        }.fold(
-            { failure -> CreateConversationResult.Failure(failure) },
-            { conversation -> CreateConversationResult.Success(conversation) }
-        )
+    suspend operator fun invoke(otherUserId: UserId): CreateConversationResult {
+        return conversationRepository.getOneToOneConversationDetailsByUserId(otherUserId)
+            .fold({ getConversationFailure ->
+                if (getConversationFailure is StorageFailure.DataNotFound) {
```

Comment:
Maybe we could transform this `if` / `else` into a `when` clause, so that we remember to update the cases when a new `StorageFailure` is added and we want to treat it differently?

Reply:
Left value from `getOneToOneConversationDetailsByUserId` is a CoreFailure, so we need to handle all subclasses of it. I don't  think we should do it right here, because we are propagating other Failures to view models

Reply:
Left value from `getOneToOneConversationDetailsByUserId` is a CoreFailure, so we need to handle all subclasses of it. I don't  think we should do it right here, because we are propagating other Failures to view models